lego/apps/restricted/serializers.py

```diff
@@ -28,6 +30,17 @@ class Meta(RestrictedMailListSerializer.Meta):
             "hide_sender",
         )
 
+    def create(self, validated_data):
+        groups = validated_data["groups"]
+        events = validated_data["events"]
+        num = reduce((lambda a, b: a + b.number_of_users), groups, 0)
+        num += reduce((lambda a, b: a + b.registration_count), events, 0)
+        if num > 500:
+            raise exceptions.ValidationError(
+                "The number of students in selected groups/arrangements exceed the permitted amount"
```

Comment:
Do you mean "arrangements" as in "events"?

Reply:
Right on 😊